plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -108,6 +115,52 @@ public ElasticsearchMetadata(TypeManager typeManager, ElasticsearchClient client
         this.schemaName = config.getDefaultSchema();
     }
 
+    private ElasticsearchColumnProperty generateProperties(IndexMetadata.Field field)
+    {
+        Map<String, ElasticsearchColumnProperty> children = ImmutableMap.of();
+        DecoderType decoderType = extractFieldDecoderType(field);
+
+        if (decoderType == DecoderType.ROW) {
+            children = ((ObjectType) field.getType()).getFields().stream()
+                    .collect(toImmutableMap(IndexMetadata.Field::getName, nestedField -> generateProperties(nestedField)));
+        }
+
+        if (decoderType == DecoderType.ARRAY) {
+            Type elementType = toTrinoType(field, field.asRawJson(), false);
+
+            if (DecoderType.of(elementType) == DecoderType.ROW) {
+                children = ((ObjectType) field.getType()).getFields().stream()
+                        .collect(toImmutableMap(IndexMetadata.Field::getName, nestedField -> generateProperties(nestedField)));
+            }
+            children = ImmutableMap.<String, ElasticsearchColumnProperty>builder()
+                    .putAll(children)
+                    .put("_elementType", new ElasticsearchColumnProperty(
+                            "elementType",
+                            children,
+                            DecoderType.of(elementType)))
+                    .build();
+        }
+
+        return new ElasticsearchColumnProperty(
+                field.getName(),
+                children,
+                extractFieldDecoderType(field));
+    }
+
+    private DecoderType extractFieldDecoderType(IndexMetadata.Field field)
```

Comment:
How do we handle `asRawJson` and `isArray` together? Like `Array(Map)` needs to be parsed as `Array(rawJson)`. In that case, we should set both properties to be set. And return decoderType as ARRAY but decode array elements as RAW_JSON. Thoughts?

Reply:
We explicitly decided not to handle that. It's confusing from a client perspective. We [throw an exception for now.](https://github.com/bitsondatadev/trino/blob/add_as_raw_json/plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/client/ElasticsearchClient.java#L579)